app/assets/src/components/views/amr_heatmap/AMRHeatmapVis.jsx

```diff
@@ -68,9 +68,16 @@ export default class AMRHeatmapVis extends React.Component {
     const genes = {};
     const alleles = {};
     sampleData.forEach(sample => {
-      sampleLabels.push({ label: sample.sample_name });
+      sampleLabels.push({ label: sample.sample_name, id: sample.sample_id });
       sample.amr_counts.forEach(amrCount => {
-        genes[amrCount.gene] = true;
+        // The following three lines are a kind of hacky workaround to the fact that
+        // the amr counts stored in the db have a gene name that includes the actual gene
+        // plus the drug class.
+        const geneNameExtractionRegex = /[^_]+/; // matches everything before the first underscore
+        const geneName = geneNameExtractionRegex.exec(amrCount.gene)[0];
```

Comment:
In this case, would it be easier to do `geneName = amrCount.gene.split("_")[0]` instead?

Reply:
Is this logic also needed in `mapAllelesToGenes`? If so, maybe it would make sense to factor out into a separate method `getGeneNameForAmrCount`.

Reply:
It actually modifies the name of the gene in each amr count in the sample data. The logic is kind of unclear though, so I'll factor it out.

Reply:
Ahh I see. 
I was expecting code in `extractLabels` to generate new objects (for the labels) but not mutate any existing data.
You might put this logic in `AMRHeatmapView` in a post-processing function right after fetching the AMR counts. `processAMRCounts`.